x-pack/plugin/esql/qa/testFixtures/src/main/resources/keep.csv-spec

```diff
@@ -540,6 +540,41 @@ c:i
 1
 ;
 
+keepSubfieldViaSuffix
+required_capability: enrich_with_qualifier
+ROW a.b.c = "abc", b.c = "bc", x.y.z = "xyz"
+| EVAL xyz_implicit = z, xyz_implicit2 = y.z, xyz_full = x.y.z, bc_full = b.c, abc_full = a.b.c
+;
+
+a.b.c:keyword | b.c:keyword | x.y.z:keyword  | xyz_implicit:keyword | xyz_implicit2:keyword | xyz_full:keyword | bc_full:keyword | abc_full:keyword
+abc           | bc          | xyz            | xyz                  | xyz                   | xyz              | bc              | abc
```

Comment:
@luigidellaquila found a good reason why implicitly addressing `x.y.z` as `y.z` or `z` is dangerous:
```
FROM index | KEEP z, y.z, x.y.z
```
As long as there's only `x.y.z` in `index`, all is good. But if the mapping of the index evolves and it gains fields `z` and `y.z.`, the output of the query will silently change.